### PR TITLE
Kotsadm

### PR DESCRIPTION
**Pull request by areed**

### CODE REVIEW

File: addons/kotsadm/0.9.4/tmpl-start-kotsadm-web.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+sed -i 's/###_GITHUB_CLIENT_ID_###/not-supported/g' /usr/share/nginx/html/index.html
+sed -i 's/###_SHIPDOWNLOAD_ENDPOINT_###/http:\/\/###_HOSTNAME_###\/api\/v1\/download/g' /usr/share/nginx/html/index.html
+sed -i 's/###_SHIPINIT_ENDPOINT_###/http:\/\/###_HOSTNAME_###\/api\/v1\/init\//g' /usr/share/nginx/html/index.html
+sed -i 's/###_SHIPUPDATE_ENDPOINT_###/http:\/\/###_HOSTNAME_###\/api\/v1\/update\//g' /usr/share/nginx/html/index.html
+sed -i 's/###_SHIPEDIT_ENDPOINT_###/http:\/\/###_HOSTNAME_###\/api\/v1\/edit\//g' /usr/share/nginx/html/index.html
+sed -i 's/###_GITHUB_REDIRECT_URI_###/http:\/\/###_HOSTNAME_###\/auth\/github\/callback/g' /usr/share/nginx/html/index.html
+sed -i 's/###_GITHUB_INSTALL_URL_###/not-supportetd/g' /usr/share/nginx/html/index.html
+sed -i 's/###_INSTALL_ENDPOINT_###/http:\/\/###_HOSTNAME_###\/api\/install/g' /usr/share/nginx/html/index.html
+
+sed -i "s/'self'/'self' ###_HOSTNAME_###/g" /etc/nginx/conf.d/default.conf
```
Comment (areed) [inline addons/kotsadm/0.9.4/tmpl-start-kotsadm-web.sh]: added for CSP

File: addons/kotsadm/0.9.4/tmpl-secret-password.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: kotsadm-password
+stringData:
+  passwordBcrypt: ${BCRYPT_PASSWORD}
```
Comment (divolgin) [inline addons/kotsadm/0.9.4/tmpl-secret-password.yaml]: Am I reading this right that this is a bcrypted password?   If it is, how is it used?

Comment (areed) [reply addons/kotsadm/0.9.4/tmpl-secret-password.yaml]: The first time the kotsadm addon is installed a password is generated and then bcrypted. The bcrypted password is used as an environment variable on the kotsadm-api pod and the user has to enter the plain password in the UI the first time.

Comment (areed) [reply addons/kotsadm/0.9.4/tmpl-secret-password.yaml]: https://github.com/replicatedhq/kots/blob/master/pkg/kotsadm/api_objects.go#L128

**Pull request merged**